proxy/http/HttpSM.cc

```diff
@@ -4622,16 +4623,16 @@ HttpSM::do_http_server_open(bool raw)
   ink_assert(pending_action == NULL);
 
   if (false == t_state.api_server_addr_set) {
-    ink_assert(t_state.current.server->port > 0);
-    t_state.current.server->addr.port() = htons(t_state.current.server->port);
+    ink_assert(t_state.current.server->dst_addr.port() > 0);
+    t_state.current.server->dst_addr.port() = htons(t_state.current.server->dst_addr.port());
```

Comment:
This will flip the byte order which seems like a bad idea. In the previous state the separate port was stored in host order and this was needed to flip it to network order to be used in the `IpEndpoint`.